File: SimpleCalculator/Form1.vb
Public Class Form1
    Dim result As Double
    Dim input As String = ""
    Dim operatorPressed As Boolean

    ' Form Load event
    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Me.Text = "Simple Calculator"
        ' Enable form to capture key presses
        Me.KeyPreview = True
    End Sub

    ' Form KeyDown event to handle keyboard input
    Private Sub Form1_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        Select Case e.KeyCode
            Case Keys.D0, Keys.NumPad0
                btnNumber_Click(btn0, Nothing)
            Case Keys.D1, Keys.NumPad1
                btnNumber_Click(btn1, Nothing)
            Case Keys.D2, Keys.NumPad2
                btnNumber_Click(btn2, Nothing)
            Case Keys.D3, Keys.NumPad3
                btnNumber_Click(btn3, Nothing)
            Case Keys.D4, Keys.NumPad4
                btnNumber_Click(btn4, Nothing)
            Case Keys.D5, Keys.NumPad5
                btnNumber_Click(btn5, Nothing)
            Case Keys.D6, Keys.NumPad6
                btnNumber_Click(btn6, Nothing)
            Case Keys.D7, Keys.NumPad7
                btnNumber_Click(btn7, Nothing)
            Case Keys.D8, Keys.NumPad8
                btnNumber_Click(btn8, Nothing)
            Case Keys.D9, Keys.NumPad9
                btnNumber_Click(btn9, Nothing)
            Case Keys.Add, Keys.Oemplus
                btnOperation_Click(btnAdd, Nothing)
            Case Keys.Subtract, Keys.OemMinus
                btnOperation_Click(btnSubtract, Nothing)
            Case Keys.Multiply
                btnOperation_Click(btnMultiply, Nothing)
            Case Keys.Divide, Keys.OemQuestion
                btnOperation_Click(btnDivide, Nothing)
            Case Keys.Enter
                btnEquals_Click(btnEquals, Nothing)
            Case Keys.Delete, Keys.Back
                btnClear_Click(btnClear, Nothing)
        End Select
    End Sub

    ' Number button click event
    Private Sub btnNumber_Click(sender As Object, e As EventArgs) _
        Handles btn0.Click, btn1.Click, btn2.Click, btn3.Click, btn4.Click,
                btn5.Click, btn6.Click, btn7.Click, btn8.Click, btn9.Click
        If operatorPressed Then
            txtDisplay.Text += " " + DirectCast(sender, Button).Text
        Else
            txtDisplay.Text += DirectCast(sender, Button).Text
        End If
        input += DirectCast(sender, Button).Text
        operatorPressed = False
    End Sub

    ' Operation button click event
    Private Sub btnOperation_Click(sender As Object, e As EventArgs) _
        Handles btnAdd.Click, btnSubtract.Click, btnMultiply.Click, btnDivide.Click
        txtDisplay.Text += " " + DirectCast(sender, Button).Text + " "
        input += " " + DirectCast(sender, Button).Text + " "
        operatorPressed = True
    End Sub

    ' Equals button click event with division by zero check
    Private Sub btnEquals_Click(sender As Object, e As EventArgs) Handles btnEquals.Click
        Try
            If input.Contains("/ 0") Then
                txtDisplay.Text = "Error: Can't divide by Zero"
                input = ""
            Else
                result = EvaluateExpression(input)
                txtDisplay.Text = result.ToString()
                input = result.ToString()
            End If
        Catch ex As Exception
            txtDisplay.Text = "Error"
            input = ""
        End Try
        operatorPressed = False
    End Sub

    ' Clear button click event
    Private Sub btnClear_Click(sender As Object, e As EventArgs) Handles btnClear.Click
        txtDisplay.Clear()
        input = ""
        result = 0
        operatorPressed = False
    End Sub

    ' Evaluate the mathematical expression
    Private Function EvaluateExpression(expression As String) As Double
        Dim table As New DataTable()
        Return Convert.ToDouble(table.Compute(expression, String.Empty))
    End Function
End Class

File: SimpleCalculator/Form1.cs
using System;
using System.Data;
using System.Windows.Forms;

namespace SimpleCalculator
{
    public partial class Form1
    {
        private double result;
        private string input = "";
        private bool operatorPressed;

        public Form1()
        {
            InitializeComponent();
        }

        // Form Load event
        private void Form1_Load(object sender, EventArgs e)
        {
            this.Text = "Simple Calculator";
            // Enable form to capture key presses
            this.KeyPreview = true;
        }

        // Form KeyDown event to handle keyboard input
        private void Form1_KeyDown(object sender, KeyEventArgs e)
        {
            switch (e.KeyCode)
            {
                case Keys.D0:
                case Keys.NumPad0:
                    {
                        this.btnNumber_Click(this.btn0, (EventArgs)null);
                        break;
                    }
                case Keys.D1:
                case Keys.NumPad1:
                    {
                        this.btnNumber_Click(this.btn1, (EventArgs)null);
                        break;
                    }
                case Keys.D2:
                case Keys.NumPad2:
                    {
                        this.btnNumber_Click(this.btn2, (EventArgs)null);
                        break;
                    }
                case Keys.D3:
                case Keys.NumPad3:
                    {
                        this.btnNumber_Click(this.btn3, (EventArgs)null);
                        break;
                    }
                case Keys.D4:
                case Keys.NumPad4:
                    {
                        this.btnNumber_Click(this.btn4, (EventArgs)null);
                        break;
                    }
                case Keys.D5:
                case Keys.NumPad5:
                    {
                        this.btnNumber_Click(this.btn5, (EventArgs)null);
                        break;
                    }
                case Keys.D6:
                case Keys.NumPad6:
                    {
                        this.btnNumber_Click(this.btn6, (EventArgs)null);
                        break;
                    }
                case Keys.D7:
                case Keys.NumPad7:
                    {
                        this.btnNumber_Click(this.btn7, (EventArgs)null);
                        break;
                    }
                case Keys.D8:
                case Keys.NumPad8:
                    {
                        this.btnNumber_Click(this.btn8, (EventArgs)null);
                        break;
                    }
                case Keys.D9:
                case Keys.NumPad9:
                    {
                        this.btnNumber_Click(this.btn9, (EventArgs)null);
                        break;
                    }
                case Keys.Add:
                case Keys.Oemplus:
                    {
                        this.btnOperation_Click(this.btnAdd, (EventArgs)null);
                        break;
                    }
                case Keys.Subtract:
                case Keys.OemMinus:
                    {
                        this.btnOperation_Click(this.btnSubtract, (EventArgs)null);
                        break;
                    }
                case Keys.Multiply:
                    {
                        this.btnOperation_Click(this.btnMultiply, (EventArgs)null);
                        break;
                    }
                case Keys.Divide:
                case Keys.OemQuestion:
                    {
                        this.btnOperation_Click(this.btnDivide, (EventArgs)null);
                        break;
                    }
                case Keys.Enter:
                    {
                        this.btnEquals_Click(this.btnEquals, (EventArgs)null);
                        break;
                    }
                case Keys.Delete:
                case Keys.Back:
                    {
                        this.btnClear_Click(this.btnClear, (EventArgs)null);
                        break;
                    }
            }
        }

        // Number button click event
        private void btnNumber_Click(object sender, EventArgs e)
        {
            if (operatorPressed)
            {
                this.txtDisplay.Text += " " + ((Button)sender).Text;
            }
            else
            {
                this.txtDisplay.Text += ((Button)sender).Text;
            }
            input += ((Button)sender).Text;
            operatorPressed = false;
        }

        // Operation button click event
        private void btnOperation_Click(object sender, EventArgs e)
        {
            this.txtDisplay.Text += " " + ((Button)sender).Text + " ";
            input += " " + ((Button)sender).Text + " ";
            operatorPressed = true;
        }

        // Equals button click event with division by zero check
        private void btnEquals_Click(object sender, EventArgs e)
        {
            try
            {
                if (input.Contains("/ 0"))
                {
                    this.txtDisplay.Text = "Error: Can't divide by Zero";
                    input = "";
                }
                else
                {
                    result = EvaluateExpression(input);
                    this.txtDisplay.Text = result.ToString();
                    input = result.ToString();
                }
            }
            catch (Exception ex)
            {
                this.txtDisplay.Text = "Error";
                input = "";
            }
            operatorPressed = false;
        }

        // Clear button click event
        private void btnClear_Click(object sender, EventArgs e)
        {
            this.txtDisplay.Clear();
            input = "";
            result = 0d;
            operatorPressed = false;
        }

        // Evaluate the mathematical expression
        private double EvaluateExpression(string expression)
        {
            var table = new DataTable();
            return Convert.ToDouble(table.Compute(expression, string.Empty));
        }
    }
}